### PR TITLE
clippy: Fix several warnings in `components/script` and `components/webgpu`

### PR DESCRIPTION
**Pull request by taniishkaaa**
Fixes several warnings in `components` as a part of #31500 

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are a part of #31500 
- [X] These changes do not require tests because they do not modify functionality, they only fix lint errors.

### CODE REVIEW

File: components/script/dom/bluetoothuuid.rs [hunk 1]
```diff
@@ -12,14 +12,14 @@ use crate::dom::bindings::reflector::Reflector;
 use crate::dom::bindings::str::DOMString;
 use crate::dom::window::Window;
 
-#[allow(clippy::upper_case_acronyms)]
```
Comment (mrobinson) [inline components/script/dom/bluetoothuuid.rs]: Doesn't removing this allow for create a new warning for the `UUID` type?

Comment (taniishkaaa) [reply components/script/dom/bluetoothuuid.rs]: it does indeed! sorry for the overlook, I'll fix it.

**Review by mrobinson — APPROVED** [review]

**Comment by mukilan** [comment]
There are some font related crashes during the merge check. They are unrelated to this PR and could potentially be due to the recent changes in font service. I'll add the PR back to merge queue to see if these crashes are intermittent.

```
  ▶ CRASH [expected OK] /encoding/legacy-mb-korean/euc-kr/euckr-decode-ks_c_5601-1987.html?6001-7000
  │ 
  │ Failed to receive a response from live font cache (thread Script(1,1), at components/fonts/system_font_service.rs:552)
  │    0: servoshell::backtrace::print
  │    1: servoshell::panic_hook::panic_hook
  │    2: std::panicking::rust_panic_with_hook
  │    3: std::panicking::begin_panic_handler::{{closure}}
  │    4: std::sys_common::backtrace::__rust_end_short_backtrace
  │    5: rust_begin_unwind
  │    6: core::panicking::panic_fmt
  │    7: fonts::system_font_service::SystemFontServiceProxy::find_matching_font_templates
  │    8: fonts::font_context::FontContext::matching_templates
  │    9: fonts::font::FontGroupFamily::members
  │   10: fonts::font::FontGroupFamily::find
  │   11: <alloc::vec::into_iter::IntoIter<T,A> as core::iter::traits::iterator::Iterator>::try_fold
  │   12: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold
  │   13: fonts::font::FontGroup::find_by_codepoint
  │   14: layout_2020::flow::inline::InlineFormattingContext::new_with_builder
  │   15: layout_2020::flow::inline::construct::InlineFormattingContextBuilder::finish
  │   16: layout_2020::flow::construct::BlockContainerBuilder<Node>::finish
  │   17: layout_2020::flow::construct::<impl layout_2020::flow::BlockContainer>::construct
  │   18: layout_2020::flow::construct::BlockLevelJob<Node>::finish
  │   19: rayon::iter::plumbing::Folder::consume_iter
  │   20: rayon::iter::plumbing::bridge_producer_consumer::helper
  │   21: <rayon::vec::IntoIter<T> as rayon::iter::IndexedParallelIterator>::with_producer
  │   22: rayon::iter::collect::collect_with_consumer
  │   23: rayon::iter::extend::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend
  │   24: layout_2020::flow::construct::BlockContainerBuilder<Node>::finish
  │   25: layout_2020::flow::construct::<impl layout_2020::flow::BlockContainer>::construct
  │   26: layout_2020::formatting_contexts::IndependentFormattingContext::construct
  │   27: layout_2020::flow::root::BoxTree::construct
  │   28: rayon_core::registry::Registry::in_worker
  │   29: layout_thread_2020::LayoutThread::handle_reflow
  │   30: <layout_thread_2020::LayoutThread as script_layout_interface::Layout>::reflow
  │   31: script::dom::window::Window::force_reflow
  │   32: script::dom::window::Window::reflow
  │   33: script::script_thread::ScriptThread::handle_msgs
  │   34: script::script_thread::ScriptThread::start
  │   35: profile_traits::mem::ProfilerChan::run_with_memory_reporting
  │   36: std::sys_common::backtrace::__rust_begin_short_backtrace
  │   37: core::ops::function::FnOnce::call_once{{vtable.shim}}
  │   38: std::sys::pal::unix::thread::Thread::new::thread_start
  │   39: <unknown>
  └   40: <unknown>
  ```

**Comment by taniishkaaa** [comment]
Thanks for merging!

**Pull request merged**